waltid-services/waltid-e2e-tests/src/test/kotlin/ExchangeExternalSignaturesApi.kt

```diff
@@ -632,33 +637,37 @@ class ExchangeExternalSignatures(private val e2e: E2ETest) {
             setBody(prepareRequest)
         }.expectSuccess()
         val prepareResponse = response.body<PrepareOID4VPResponse>()
-        val submitResponse = client.post("/wallet-api/wallet/$walletId/exchange/external_signatures/presentation/submit") {
-            setBody(
-                SubmitOID4VPRequest.build(
-                    prepareResponse,
-                    disclosures = if (addDisclosures) matchedCredentialList.filter { it.disclosures != null }.associate {
-                        Pair(
-                            it.id, listOf(
-                                if (forgeDisclosures) forgeSDisclosureString(it.disclosures!!) else it.disclosures!!
-                            )
-                        )
-                    } else null,
-                    w3cJwtVpProof = prepareResponse.w3CJwtVpProofParameters?.let { params ->
-                        holderKey.signJws(
-                            params.payload.toJsonElement().toString().toByteArray(),
-                            params.header,
-                        )
-                    },
-                    ietfSdJwtVpProofs = prepareResponse.ietfSdJwtVpProofParameters?.map { params ->
-                        IETFSdJwtVpTokenProof(
-                            credentialId = params.credentialId, sdJwtVc = params.sdJwtVc, vpTokenProof = holderKey.signJws(
+        val submitResponse =
```

Comment:
this looks similar to the one from above (line:530)
perhaps worth extracting it into a method?